### PR TITLE
feat(cli): interactive `inplan login` via the browser handoff

### PR DESCRIPTION
**Pull request by melly-lgtm**
## What

Makes `inplan login` actually usable by a human. Today a bare `inplan login` only prints a usage error demanding raw `--url/--anon/--refresh` tokens that a user has no way to obtain — so the in-product **"Wait for my local agent"** instructions ("Check the CLI… sign in with `inplan login`… then `inplan wait --remote <id>`") dead-end at the sign-in step (found in manual prod QA, "M4-FAIL-1").

This adds the interactive browser handoff — the same flow the desktop app already uses, whose web `/cli-auth` page already exists in inplan-cloud:

1. `inplan login` (no args) spins up a one-shot `127.0.0.1` listener and opens `https://inplan.ai/cli-auth?port=<port>&state=<nonce>` in the browser.
2. The user signs in there (Google / GitHub / email).
3. The page POSTs the project url/anon + a fresh refresh token back over the loopback; the CLI validates `state`, persists via the existing `saveAuth`, and prints `{ "status": "logged_in" }`.

The token only ever crosses localhost; the random `state` nonce is echoed back so a stray local request can't inject a session.

## Backward compatible

The flag/env form is preserved for scripts and the desktop app: when **all** of `--url/--anon/--refresh` (or `INPLAN_SUPABASE_URL` / `INPLAN_SUPABASE_ANON_KEY` / `INPLAN_REFRESH_TOKEN`) are present, it stores them directly without opening a browser. `$INPLAN_WEB_URL` overrides the default `https://inplan.ai` (self-hosted / dev).

## Files

- `packages/cli/src/cliLogin.ts` (new): `browserLogin()` — loopback listener (CORS-preflight aware, 64 KB body cap, state-validated), OS browser opener, resolves with the credentials to persist. `open` is injectable for tests.
- `packages/cli/src/cli.ts`: `doLogin` is interactive by default, non-interactive when fully specified; usage text updated.

## Testing

- New `packages/cli/test/cliLogin.test.ts` (5 tests): handoff happy-path; the opened URL's origin/port/state shape; mismatched-state POST rejected with 403 while the real handoff still completes; malformed body → 400; timeout. **All green.**
- Full suite via the pre-commit gate: **655 passed / 2 skipped**, coverage **96.05% ≥ 95%**.

## Rollout note

Reaches users after the CLI is republished to npm (the normal release step). The web `/cli-auth` half is already live in inplan-cloud. Pairs with the inplan-cloud auth-UX PR.

<a href="https://cubic.dev/pr/melly-lgtm/inplan/pull/15?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Interactive browser-based CLI login via system browser with local callback support
  * Login command now supports both interactive browser flow and non-interactive credential flags

* **Bug Fixes / Behavior**
  * Non-interactive credentials accepted only when all required flags/env are present; interactive flow used otherwise
  * Interactive login failures now report an error and exit with a non-zero status

* **Tests**
  * Added tests covering browser-based authentication and edge cases (timeouts, malformed handoffs)

### CODE REVIEW

File: packages/cli/src/cliLogin.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+/** Best-effort: open `url` in the OS browser. Errors are swallowed — the URL is also
+ *  printed so the user can open it by hand. */
+function openInBrowser(url: string): void {
+  const cmd = process.platform === "darwin" ? "open" : process.platform === "win32" ? "cmd" : "xdg-open";
+  const args = process.platform === "win32" ? ["/c", "start", "", url] : [url];
+  try {
+    spawn(cmd, args, { detached: true, stdio: "ignore" }).unref();
+  } catch {
+    /* opener missing (headless/CI) — the printed URL is the fallback */
+  }
```
[the remark this answers is not in the document]
Comment (melly-lgtm) [reply packages/cli/src/cliLogin.ts:55]: Fixed in 4bf3937 — added a child `error` listener so a missing opener surfaces as the printed-URL fallback instead of an unhandled async error event.

File: packages/cli/src/cliLogin.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+function isHandoff(v: unknown): v is Handoff {
+  if (typeof v !== "object" || v === null) return false;
+  const o = v as Record<string, unknown>;
+  return typeof o.state === "string" && typeof o.url === "string" && typeof o.anon === "string" && typeof o.refresh === "string";
+}
```
[the remark this answers is not in the document]
Comment (melly-lgtm) [reply packages/cli/src/cliLogin.ts:68]: Fixed in 4bf3937 — `isHandoff` now requires `email` to be a string when present (`{ email: {} }` is rejected), with a regression test.
